microsoft-edge/devtools-guide-chromium/memory-problems/heap-snapshot-schema.md

```diff
@@ -0,0 +1,209 @@
+
+So this array should have `node_count` * `snapshot.metadata.node_fields.length` total elements. To re-construct a node you need to read a chunk of 7 elements from this array.
+
+`node_fields` | Description  
+:- | :- 
+type | See `node_types[0]` to find out what are the possible types
+name | A string 
+id | A number 
+self_size | Size in bytes
+edge_count | Number of edges for this node
+trace_node_id | - 
+detachedness | Number, 0 or 1 if the object can be reached from the window
+
+`node_types` | Description  
+:- | :- 
+Hidden | V8 internal elements that doesn't correspond directly to a user-controllable JavaScript object. In DevTools, these all show up under the category name "(system)". Even though these objects are internal, they can be an important part of retainer paths.
+Object | Any user-defined `{x:2}` or new `Foo(4)`. Contexts, which show up in DevTools as "system / Context" hold variables that had to be allocated on the heap because they're used by a nested function.
+Native | Things allocated by Blink, not by V8. Mostly DOM: HTMLDivElement, CSSStyleRule, etc. 
+Concatenated string | The result of concatenating two strings with the "+" operator. Rather than making a new string with a copy of all the data from the two source strings, V8 creates a ConsString object with pointers to the two source strings. From a JS perspective, it acts just like any other string, but from a memory profiling perspective, it is different.
+Sliced string | The result of a substring operation, such as using String.prototype.substr or String.prototype.substring. V8 avoids copying string data by instead creating a SlicedString which points to the original string and specifies the start index and length. From a JS perspective, it acts just like any other string, but from a memory profiling perspective, it is different.
+Array | Various internal lists, which show up in the devtools with the category name "(array)". Like Hidden, this category clumps together a bunch of unrelated things. Many of the objects here are named "(object properties)" or "(object elements)", indicating that they contain the string-keyed or numeric-keyed properties of a JavaScript object.
+Code | Things that grow proportionally to the amount of script, and/or the number of times that functions run..  
+Synthetic | Synthetic nodes don't correspond to anything actually allocated in memory. As far as I can tell, these are used only to distinguish the different kinds of GC roots.
+
+
+## Edges
+
+Similar to the **Nodes** array, edges contains all the individual elements needed to re-construct the edges and we can calculate the number of elements of this array by multiplying `edge_count` * `metadata.edge_fields.length`.
+
+Edges are stored sequentially and to read them properly you need to parse the nodes first, because each node knows how many edges it has. 
```

Comment:
Reading this sentence makes me think that we might need a screenshot somewhere that explains how the data is organized in the file. This would be useful for visual learners. This sentence is quite hard to parse and a picture can be worth a thousand words.
Do you think there's an easy way to "draw" a little heapsnapshot sample to make things clearer?